### ISSUE TITLE
double free bug in `HaikuOutput::~HaikuOutput()`

### ISSUE THREAD

**Issue by MaxKellermann** [MusicPlayerDaemon/MPD#184]
The semaphores get allocated in `Open()` and freed in `Close()`, but they are freed again in the destructor.

**Comment by MaxKellermann** [comment]
This is a rather serious bug. Will you fix it, or shall I declare MPD's Haiku support unmaintained?

**Comment by MaxKellermann** [comment]
MPD's Haiku support is now unmaintained: 7de8fd04a4dfbd0dbc1d021f3d096724069cde2b
The plugin will be removed soon.

**Comment by mmuman** [comment]
No it's not. It's just that things keep piling up here, plus last month my ISP decided I shouldn't get mails from github…

I even have pending patches around it seems…

But now when I try to compile…

```
meson.build:1:0: ERROR:  Meson version is 0.48.2 but project requires >= 0.49.0.
```

So I'll first have to update our meson package. Oh, dependency hell 😁

**Comment by mmuman** [comment]
Actually… this bug is not that serious.

Semaphores are kernel-managed objects, like file descriptors, calling delete_sem() twice is as dangerous as calling close() twice on an fd, it would just return -1 with errno=EINVAL. In no way would it screw up the heap or something.

**Comment by MaxKellermann** [comment]
0d8942e64a0e1ef860e31a81246a0d17bbfda075

Btw. calling `close()` twice on a file descriptor is a very serious bug! It is very likely that another part of the process has meanwhile allocated the same file descriptor. Calling `close()` can have serious side effects, including data corruption.

**Comment by mmuman** [comment]
Yes but no crash like a double-free would do, at least if error handling is done. Anyway.

**Comment by mmuman** [comment]
Btw, should we work on master on that 0.22 branch?

**Comment by MaxKellermann** [comment]
On 2019/03/30 03:56, François Revol <notifications@github.com> wrote:
> Btw, should we work on master on that 0.22 branch?

Bug fixes to v0.21.x, please.
